core/spec.md

```diff
@@ -5472,6 +5524,48 @@ as defined below:
   - `true`
   - `false`
 
+##### `ancestor` Attribute
+- Type: String
+- Description: The `versionid` of this Version's ancestor.
+
+  The `ancestor` attribute MUST be set to the `versionid` of this Version's
+  ancestor If this Version is a root of an ancestor hierarchy tree then it
+  MUST be set to its own `versionid` value.
+
+  When creating a Version without explicitly setting the `ancestor`
+  attribute, the server MUST set the `ancestor` to the most recent Version's
+  `versionid` attribute. If no Versions exist, the `ancestor` attribute MUST
+  be set to the `versionid` of the Version being created, making it a root.
+  If there are existing Versions, the strategy for finding the most recent
+  Version is as follows:
+  1. Find all Versions that are not referenced as an `ancestor` in other
+     Versions.
+  1. If many exist, find the Version with the most recent `createdat` date.
+  1. If many exist, select the first Version sorting all Versions in
+     descending alphabetical order based on the `versionid` attribute.
+
+  If a write operation contains multiple Versions with the `ancestor` attribute
+  omitted, the server MUST order all of those Versions based on the `createdat`
+  attribute and then alphabetically (ascending). The first Version will have
+  the most recent Version's `versionid` as its `ancestor` as clarified above.
+
+  When deleting a Version, the server MUST update the `ancestor` attribute
+  of any Version that points to the deleted Version to point to itself,
+  making it a new root.
+
+  If a create operation asks the server to choose the `versionid` when
+  creating a root Version, the `versionid` is not yet known and therefore
+  cannot be assigned the value in the `ancestor` attribute. In those cases a
+  value of `request` MAY be used as a way to reference the Version being
```

Comment:
I think this needs to be a MUST since we're already within the scope of "creating a root with an unknown id", so I don't think there are alternative available... they MUST use "request"